### PR TITLE
Feature/skale 4698 Add interface between skaled and skale-admin

### PR DESCRIPTION
**Pull request by dmytrotkk**
Integration of the `skaled.status` file for sChain containers:

- [x] Add handler in RPC checks - do not kill skaled if snapshot is downloading
- [x] Replace `.rotation` file with status file to determine post-rotation mode
- [x] Add API to request statuses for all sChains on the node

### CODE REVIEW

File: core/schains/skaled_status.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    @property
+    def is_downloading_snapshot(self):
+        subsystem_running = self.subsystem_running
+        if not subsystem_running:
+            return
+        return subsystem_running['SnapshotDownloader']
+
+    @property
+    def is_exit_time_reached(self):
```
Comment (badrogger) [inline core/schains/skaled_status.py]: Suggest to rename to exit_time_reached or make it a function.

File: tests/conftest.py [hunk 2]
```diff
@@ -78,6 +85,22 @@ def get_random_string(length=8):
...
+            "Blockchain": False,
+            "Rpc": False
+        },
+        "exitState": {
+            "ClearDataDir": False,
+            "StartAgain": False,
+            "StartFromSnapshot": False,
+            "ExitTimeReached": exit_time_reached
+        }
+    }.copy()
```
Comment (badrogger) [inline tests/conftest.py]: Why do we need copy here ?

**Review by badrogger — CHANGES_REQUESTED** [review]

File: tests/conftest.py (hunk 2 above)
Comment (dmytrotkk) [reply tests/conftest.py]: removed

File: core/schains/skaled_status.py (hunk 1 above)
Comment (dmytrotkk) [reply core/schains/skaled_status.py]: renamed

File: core/schains/monitor/rpc.py [hunk 3]
```diff
@@ -45,8 +46,12 @@ def monitor_schain_rpc(
         logger.warning(f'{schain_name} RPC monitor failed: container doesn\'t exit')
         return
 
-    if is_exited_with_zero(schain_name, dutils=dutils):
-        logger.info(f'{schain_name} container exited with zero, skipping RPC monitor')
+    if skaled_status.exit_time_reached:
```
Comment (badrogger) [inline core/schains/monitor/rpc.py:49]: I think It's better to reset rpc restart counter in both conditions

File: core/schains/monitor/rpc.py [hunk 4]
```diff
@@ -45,8 +46,12 @@ def monitor_schain_rpc(
...
+        logger.info(f'{schain_name} - Skipping RPC monitor: exit time reached')
+        return
+
+    if skaled_status.downloading_snapshot:
```
Comment (badrogger) [inline core/schains/monitor/rpc.py:54]: Not sure if it's expected, but monitor_container from core.schains.runner still relies on container exit codes. Should it be changed?

**Review by badrogger — CHANGES_REQUESTED** [review]

File: core/schains/monitor/rpc.py (hunk 3 above)
Comment (dmytrotkk) [reply core/schains/monitor/rpc.py:49]: added

File: core/schains/monitor/rpc.py (hunk 4 above)
Comment (dmytrotkk) [reply core/schains/monitor/rpc.py:54]: you mean `is_schain_container_failed(schain_name, dutils=dutils)` from `core.schains.monitor.containers`?

Comment (dmytrotkk) [reply core/schains/monitor/rpc.py:54]: fixed

**Review by badrogger — DISMISSED** [review]

**Pull request merged**